### ISSUE TITLE
Create a compiled library file for Titan runtime

### ISSUE THREAD

**Issue by gligneul** [titan-lang/titan-v0#89]
There is some runtime code embedded to the current compiler output. We should have a single `.so`/`.a` for the code shared between different Titan modules.
A solution for this issue will also fix #87.

**Comment by mascarenhas** [comment]
I do not think this runtime should be a shared library, its size will probably not be big enough to offset the extra indirection. There is also the question of where the source for this runtime (and its compiled object file) is installed, and how the Titan compiler finds them (without inconveniencing the user).

**Comment by hugomg** [comment]
I think the least inconvenient place to put the runtime would be bundled together with our version of Lua. (Yes, this would come at the cost of making it harder to run titan with an unmodified Lua)

**Comment by gligneul** [comment]
This will only become a real problem when the runtime grow. Do we have any plans for a Titan standard library?

**Comment by mascarenhas** [comment]
I think the standard library can just be a collection of regular modules, loaded by the standard module loading mechanism.
